src/sentry/rules/processor.py

```diff
@@ -143,16 +145,24 @@ def bulk_get_rule_status(self, rules: Sequence[Rule]) -> Mapping[int, GroupRuleS
         return rule_statuses
 
     def condition_matches(
-        self, condition: Mapping[str, Any], state: EventState, rule: Rule
+        self, condition: dict[str, Any], state: EventState, rule: Rule
     ) -> bool | None:
         condition_cls = rules.get(condition["id"])
         if condition_cls is None:
             self.logger.warning("Unregistered condition %r", condition["id"])
             return None
 
-        condition_inst: EventCondition = condition_cls(self.project, data=condition, rule=rule)
+        condition_inst = condition_cls(self.project, data=condition, rule=rule)
+        if not isinstance(condition_inst, EventCondition) and not isinstance(
+            condition_inst, EventFilter
+        ):
```

Comment:
this can be written slightly more succinctly:  `if not isinstance(condition_inst, (EventCondition, EventFilter)): ...`